Check user rights to create content to give access to main bottom sheet

develric opened the pull request:
Fixes #11388

When a user is an Author/Contributor of a site we already remove the access to pages in My Site screen (both Quick Action and List row item):

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/76911198-80a43780-68b0-11ea-9669-399819b6e370.png>
</p>

This PR extends that check also to the FAB opening directly the Editor for post creation when a user is a User/Contributor (or not a Self-Hosted site Admin).

cc FYI @rachelmcr and  @designsimply.

## To test:
In the following the test must be repeated both with a WP.com and a Self-Hosted site.

- Add a user with Author/Contributor role to a site (you can do it for example from the web)
- Login in the app with that user and switch to the site
- Note that pages access are removed like the image above
- Tap on the Fab and check you get the editor opened for post editing
- Create the post and publish it; note you get it published in the posts list
- Change the User role to be an Editor/Administrator (you can do it for example from the web)
- Restart the app and check the desired site is selected
- Note you now have access to pages content
- Tap on the Fab and check you get the bottom sheet for post/page creation
- Test you can create both post and page and get it uploaded and reported in the post/page lists
- Clean the app cache and login again to show the FAB tooltip. Check that on a site with Editor/Administrator rights you get `Create a post or page` message where on a site with Author/Contributor rights you get `Create a post` message.
- Long press on the FAB to get a toast. Check that on a site with Editor/Administrator rights you get `Create a post or page` message where on a site with Author/Contributor rights you get `Create a post` message.
- Use Talkback to get the description of the FAB. Check that on a site with Editor/Administrator rights you get `Create a post or page` message where on a site with Author/Contributor rights you get `Create a post` message.

Hi @osullivanchris 👋, if you want/can have a look into it I think you can use the apk generated in this PR for checking.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

planarvoid commented:
hey @develric, it looks really good! I've found one thing. When I checked the site as a contributor, I still saw the tooltip saying "Create a post or page" which should be gone or changed. 

![Screenshot_1584617750](https://user-images.githubusercontent.com/1079756/77063783-957ee900-69de-11ea-9af2-ecfab72881c7.png)

develric commented:
Hey @planarvoid 👋. Good call 😊! Looking again into it I also noticed we need to manage the FAB long pressed and Description message for accessibility. 

I should have addressed it now. 

Not sure it is the best approach, I added a `mViewModel.onResume` method in the `onResume` of the `WPMainActivity` to get the tooltip message updated when switching between a site to which the user has full access and a site to which the user has not full access. Added 3 test steps in the PR description related to the additions.
